### PR TITLE
fix(@aws-amplify/auth): check if has the right scope to send the request GetUserData

### LINKED ISSUE #2861: How to disable calls to `AWSCognitoIdentityProviderService.GetUser` when `aws.cognito.signin.user.admin` is disabled?
** Which Category is your question related to? **
Auth

** What AWS Services are you utilizing? **
Cognito User Pool

** Provide additional details e.g. code snippets **

I searched the issues before posting this, but am still looking for an answer. It seems that the amplify javascript sdk will always make 2 calls to `POST https://cognito-idp.{region}.amazonaws.com` using the `X-Amz-Target: AWSCognitoIdentityProviderService.GetUser` header after a successful user login, and we would like to disable these. They succeed when the `aws.cognito.signin.user.admin` oauth scope is allowed, but when disallowed, they return a 400 Bad Response.

We have the oauth scope in question disabled and want to keep it disabled. I found a similar question in #1262, but the answer only says it is "expected behavior" with no explanation as to why. Our app functions perfectly fine when the scope is disabled and the requests error out. Can we disable the calls? If so how, and if not, why?

I also read #1906, #2251, and #1813, and updated our aws-amplify dependency to 1.1.22, but still seeing the same behavior.

### PR DESCRIPTION
**Pull request by powerful23**
*Issue #, if available:*
fixes #2861 
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/__tests__/auth-unit-test.ts [hunk 1]
```diff
@@ -2086,6 +2127,54 @@ describe('auth unit test', () => {
...
+                    return new CognitoAccessToken({AccessToken: 'accessToken'});
+                });
+            
+            const spyon5 = jest.spyOn(CognitoAccessToken.prototype, 'decodePayload')
+                .mockImplementation(() => {
+                    return { scope: '' };
+                });
+            
+            expect.assertions(1);
+            expect(await auth.currentUserPoolUser()).toBe(user);
```
Comment (elorzafe) [inline packages/auth/__tests__/auth-unit-test.ts]: Just to make it more explicit I would make the test fail if the function user.getUserData() was called more than 0 times. Could be an edge case that make the test pass and still tries to call the service.

File: packages/auth/src/Auth.ts [hunk 2]
```diff
@@ -937,39 +938,47 @@ export default class AuthClass {
...
+                                const attributes = that.attributesToObject(attributeList);
+                                Object.assign(user, { attributes, preferredMFA });
+                                return res(user);
+                            },
+                            { bypassCache }
+                        );
+                    } else {
+                        logger.debug(`Unable to get the user data because the ${USER_ADMIN_SCOPE} ` + 
+                            `is not in the scopes of the access token`);
+                        return res(user);
```
Comment (elorzafe) [inline packages/auth/src/Auth.ts]: Should this be rejected in this case? because you are trying to get the attributes, and because of scope is not on the token you cannot get it?

Comment (powerful23) [reply packages/auth/src/Auth.ts]: The function is to return the current user pool user. Although the user can't get attributes, but he's still logged in. So we should still return the user, same logic like here: https://github.com/aws-amplify/amplify-js/blob/master/packages/auth/src/Auth.ts#L951

File: packages/auth/__tests__/auth-unit-test.ts (hunk 1 above)
Comment (powerful23) [reply packages/auth/__tests__/auth-unit-test.ts]: Good catch! I added an assertion to check if `getUserData` is called or not.
